### ISSUE TITLE
[CT-1414] [Bug] Models with upper case fail in consecutive run

### ISSUE THREAD

**Issue by vargacypher** [dbt-labs/dbt-core#6151]
### Current Behavior

When running (dbt run) the model with an uppercase name, the first run succeeds, but the consecutive run fails.

### Expected Behavior

I expected my models to either be executed successfully also when re-running them

What happened is the initial run succeeded, but the consecutive runs failed with an error.
 - When searching for a relation, dbt found an approximate match. Instead of guessing
- which relation to use, dbt will move on. Please delete ............, or rename it to be less ambiguous.

### Steps To Reproduce

I create an incremental model  named contacts_test and then run it, first with lowercase name and then with uppercase.
```
{{ config(
    materialized="incremental",
    order_by='(ID)',
    engine='ReplacingMergeTree()',
    inserts_only=true,
    post_hook='OPTIMIZE TABLE {{this}} FINAL DEDUPLICATE ;'
) }}

with blabla as (
    SELECT TOP 100 *
    FROM {{ source('test', 'test') }} 
)
select *
from blabla

{% if is_incremental() %}

where SOURCE_CREATED_DT > (select max(SOURCE_CREATED_DT) from {{ this }})

{% endif %}

```
### Relevant log output

```shell
*LOWERCASE Models:*

First run:

dbt run -m models/marts/contacts_test.sql --vars '{"database":"****","profile":"general"} '
16:58:17  Running with dbt=1.3.0
16:58:20  Found 34 models, 2 tests, 0 snapshots, 0 analyses, 366 macros, 0 operations, 0 seed files, 15 sources, 0 exposures, 
0 metrics
16:58:26  Concurrency: 1 threads (target='dev')
16:58:26  
16:58:26  1 of 1 START sql incremental model ****.contacts_test ................ [RUN]
16:58:27  1 of 1 OK created sql incremental model ****.contacts_test ........... [OK in 1.09s]
16:58:27  
16:58:27  Finished running 1 incremental model in 0 hours 0 minutes and 7.04 seconds (7.04s).
16:58:27  
16:58:27  Completed successfully
16:58:27
16:58:27  Done. PASS=1 WARN=0 ERROR=0 SKIP=0 TOTAL=1


Second run:


dbt run -m models/marts/contacts_test.sql --vars '{"database":"****","profile":"general"} '
16:58:52  Running with dbt=1.3.0
16:58:56  Found 34 models, 2 tests, 0 snapshots, 0 analyses, 366 macros, 0 operations, 0 seed files, 15 sources, 0 exposures, 
0 metrics
16:59:03  Concurrency: 1 threads (target='dev')
16:59:03
16:59:03  1 of 1 START sql incremental model ****.contacts_test ................ [RUN]
16:59:04  1 of 1 OK created sql incremental model ****.contacts_test ........... [OK in 1.48s]
16:59:04
16:59:04  Finished running 1 incremental model in 0 hours 0 minutes and 8.34 seconds (8.34s).
16:59:05
16:59:05  Completed successfully
16:59:05
16:59:05  Done. PASS=1 WARN=0 ERROR=0 SKIP=0 TOTAL=1


*UPPERCASE Models:*


First run:

dbt run -m models/marts/CONTACTS_TEST.sql --vars '{"database":"****","profile":"general"} ' 
16:54:24  Running with dbt=1.3.0
16:54:24  Unable to do partial parsing because a project config has changed
16:54:29  Found 34 models, 2 tests, 0 snapshots, 0 analyses, 366 macros, 0 operations, 0 seed files, 15 sources, 0 exposures, 
0 metrics
16:54:36  Concurrency: 1 threads (target='dev')
16:54:36  
16:54:36  1 of 1 START sql incremental model ****.CONTACTS_TEST ................ [RUN]
16:54:37  1 of 1 OK created sql incremental model ****.CONTACTS_TEST ........... [OK in 1.16s]
16:54:37  
16:54:37  Finished running 1 incremental model in 0 hours 0 minutes and 7.96 seconds (7.96s).
16:54:37  
16:54:37  Completed successfully
16:54:37
16:54:37  Done. PASS=1 WARN=0 ERROR=0 SKIP=0 TOTAL=1


Second run:

dbt run -m models/marts/CONTACTS_TEST.sql --vars '{"database":"****","profile":"general"} '
16:54:58  Running with dbt=1.3.0
16:55:02  Found 34 models, 2 tests, 0 snapshots, 0 analyses, 366 macros, 0 operations, 0 seed files, 15 sources, 0 exposures, 
0 metrics
16:55:08  Concurrency: 1 threads (target='dev')
16:55:08  
16:55:08  1 of 1 START sql incremental model ****.CONTACTS_TEST ................ [RUN]
16:55:08  1 of 1 ERROR creating sql incremental model ****.CONTACTS_TEST ....... [ERROR in 0.04s]
16:55:08  CANCEL query model.****.CONTACTS_TEST ........................................... [CANCEL]
16:55:08
16:55:08  Compilation Error in model CONTACTS_TEST (models/marts/CONTACTS_TEST.sql)
16:55:08    When searching for a relation, dbt found an approximate match. Instead of guessing
16:55:08    which relation to use, dbt will move on. Please delete ****.CONTACTS_TEST, or rename it to be less ambiguous.
16:55:08    Searched for: ****.contacts_test
16:55:08    Found: ****.CONTACTS_TEST
16:55:08
16:55:08    > in macro is_incremental (macros/materializations/models/incremental/is_incremental.sql)
16:55:08    > called by model CONTACTS_TEST (models/marts/CONTACTS_TEST.sql)
16:55:09  Encountered an error:
FailFast Error in model CONTACTS_TEST (models/marts/CONTACTS_TEST.sql)
  Failing early due to test failure or runtime error
```

### Environment

```markdown
- OS:CentOS 7
- Python:3.8.10
- dbt:>=1.2.2 (tested with 1.3.0 too)
- dbt-clickhouse:1.2.2
```

### Which database adapter are you using with dbt?

other (mention it in "Additional Context")

### Additional Context

clickhouse

**Comment by lostmygithubaccount** [comment]
I'm unable to simply reproduce this on 1.3 with Snowflake, using an incremental model with no real data. it's unclear if this is specific to the clickhouse adapter, in which case I'd recommend opening the issue there @vargacypher https://github.com/ClickHouse/dbt-clickhouse/issues

**Comment by jtcohen6** [comment]
Agree! This may have to do with the fact that `dbt-clickhouse` (like `dbt-snowflake`) unquotes everything by default: https://github.com/ClickHouse/dbt-clickhouse/blob/b0f33aee8191ccd38cf7cac23e72ebdfe835255f/dbt/adapters/clickhouse/relation.py#L7-L11

I'm going to close this for now, as it's unlikely to be a problem in `dbt-core`

**Comment by vargacypher** [comment]
Hi guys, thanks for the response.

The problem will be fixed in the latest 1.3 release of dbt-clickhouse.
https://github.com/ClickHouse/dbt-clickhouse/issues/110

Again, thanks a lot for the support <3
